fix: transpile typescript enums

bitpshr opened the pull request:
This pull request adds a new module to transpile TypeScript enums.

Resolves #534

**Examples:**

1. **Without initializers**
    
    Input:
    ```
    enum Test { a, b }
    ```
    Output:
    ```
    const Test = {a: 0, 0: "a", b: 1, 1: "b"}
    ```

1. **With an explicit numeric initializer**
    
    Input:
    ```
    enum Test { a = 3, b }
    ```
    Output:
    ```
    const Test = {a: 3, 3: "a", b: 4, 4: "b"}
    ```

1. **With heterogenous initializers**
    
    Input:
    ```
    enum Test { a = 'foo', b = 0, c }
    ```
    Output:
    ```
    const Test = {a: "foo", b: 0, 0: "b", c: 1, 1: "c"}
    ```

ooflorent commented:
The output is wrong.
Given the following enum:
```ts
enum Test { a, b }
```

The output should be:
```js
const Test = {
  a: 0,
  b: 1,
  0: "a",
  1: "b",
}
```

ooflorent commented:
I was wondering if you we should not transpile `const enum` to regular ones Rome could compile them too. Please also not that TypeScript compiles these 2 enums the same way:

```ts
enum Test1 { a, b }
enum Test1 { a = 0, b = 1 }
```

The presence of the index properties is for numeric values, not initializers.

bitpshr commented:
Thanks for the feedback @ooflorent, and good catch re: the semantics of index properties on the resulting object. I updated this logic and updated the PR description with clearer input / output examples.

This should be ready for another pass when you have time.

bitpshr commented:
@ooflorent how could we properly handle the TypeScript const enum expressions, e.g. `B = 1 + 2`? The TypeScript compiler computes a subset of values at compile time, but Rome does not. Should this transpile function really traverse the `initializer` AST and manually perform computation?
